test/test_queue.rb

```diff
@@ -24,4 +24,35 @@ def test_flaky_jobs
 
     assert_failures(["./spec/legit_failure_spec.rb[1:3]"], queue)
   end
+
+  def test_fail_fast
+    build_id = rand_id
+
+    Process.wait(start_worker(
+      build_id: build_id, suite: "failing_suite", extra_args: "--fail-fast 1"))
+
+    queue = RSpecQ::Queue.new(build_id, "foo", REDIS_HOST)
+
+    assert_queue_well_formed(queue)
+
+    assert queue.build_failed_fast?
+    refute queue.build_successful?
+    assert queue.example_failures.keys.length +
+           queue.non_example_errors.keys.length >=
+           queue.fail_fast
+  end
+
+  def test_no_fail_fast
```

Comment:
This test is also redundant, the same thing is implicitly tested by all other cases. We can just remove it.